cherab/core/model/plasma/recombination.pyx

```diff
@@ -47,6 +47,22 @@ cdef class RecombinationLine(PlasmaModel):
         # ensure that cache is initialised
         self._change()
 
+    def __getstate__(self):
+        state = super().__getstate__()
+        state["line"] = self._line
+        state["lineshape_class"] = self._lineshape_class
+        state["lineshape_args"] = self._lineshape_args
+        state["lineshape_kwargs"] = self._lineshape_kwargs
+        return state
+    
+    def __setstate__(self, state):
+        super().__setstate__(state)
+        self._line = state["line"]
+        self._lineshape_class = state["lineshape_class"]
+        self._lineshape_args = state["lineshape_args"]
+        self._lineshape_kwargs = state["lineshape_kwargs"]
+        self._change()
```

Comment:
This assumes that `lineshape_args` and `lineshape_kwargs` are pickle-able objects. For example, for `ZeemanMultiplet`  that's not true, because it initializes with the `ZeemanStructure` instance, which contains `Function1D` objects.

Reply:
damn, I din't check the upcoming changes. Well I will look at what could be done, but I'm not optimistic :-/

Reply:
I think `Interpolator1DArray` will be used in `ZeemanStructure` in 99% of cases. It cannot be pickled now, but this can be fixed if needed. But there is no guarantee that any other child of `Function1D` will not be used.

Reply:
This PR was also motivated by the increasing pressure on data provenance. This was supposed to help a bit. If it is not possible due to the accepted changes, lets not push it. The provenance can be achieved with other solutions.

Reply:
I was going to propose adding `is_pickleable` class attribute to `LineShapeModel`, so `__setstate__` could raise RuntimeError when trying to pickle a `PlasmaModel` with non-pickle-able line shape. Are you sure you want to close it?